static/js/containers/ChannelFollowControls.js

```diff
@@ -0,0 +1,121 @@
+/* global SETTINGS: false */
+// @flow
+import React from "react"
+import R from "ramda"
+import { connect } from "react-redux"
+
+import DropdownMenu from "../components/DropdownMenu"
+
+import { actions } from "../actions"
+import { hideDropdown, showDropdown } from "../actions/ui"
+import { leaveChannel } from "../util/api_actions"
+
+import type { Dispatch } from "redux"
+import type { Channel } from "../flow/discussionTypes"
+
+export const CHANNEL_FOLLOW_DROPDOWN = "CHANNEL_FOLLOW_DROPDOWN"
+
+type Props = {
+  channel: Channel,
+  isDropdownOpen: boolean,
+  showDropdown: () => void,
+  hideDropdown: () => void,
+  followChannel: () => void,
+  unfollowChannel: () => void,
+  leaveChannel: () => void,
+  loadChannel: () => void
+}
+
+export const DropDownArrow = () => (
```

Comment:
We have an identical definition in `UserMenu`, could you move these into a utility component file instead?

Reply:
Good catch. Meant to do this earlier